FV3GFS/FV3GFS_io.F90

```diff
@@ -3087,12 +3088,20 @@ subroutine gfdl_diag_register(Time, Sfcprop, Gfs_diag, Cldprop, &
     Diag(idx)%mod_name = 'gfs_phys'
     Diag(idx)%cnvfac = cn_one
     Diag(idx)%time_avg = .TRUE.
-    Diag(idx)%time_avg_kind = 'rad_sw'
+    if (.not. override_surface_radiative_fluxes) then
+      Diag(idx)%time_avg_kind = 'rad_sw'
+    endif
     Diag(idx)%intpl_method = 'bilinear'
     allocate (Diag(idx)%data(nblks))
-    do nb = 1,nblks
-      Diag(idx)%data(nb)%var2 => Gfs_diag(nb)%fluxr(:,3)
-    enddo
+    if (override_surface_radiative_fluxes) then
```

Comment:
Could use a bit more explanation of what's happening here as well as the few blocks below where you are choosing between a named variable and one of the `fluxr` constituents.

Reply:
Good idea.  I'll add a comment each place it occurs.